### DISCUSSION TITLE
GitHub-like activity heat chart

### DISCUSSION THREAD

**Discussion by vnugent** [recharts/recharts#3682] [Q&A]
Is it possible to create a similar chart to Github activity heat chart?  Maybe customize stack bars to show days of the week.

---

<img width="924" alt="Screenshot 2023-08-04 at 9 56 12 AM" src="https://github.com/recharts/recharts/assets/3805254/d9d5d3b7-f76c-492d-ab90-5d3b4e9e5700">

**Comment by ckifer** [comment]
Its probably worth a different chart type to be honest. Implementing this in a bar or stacked bar wouldn't be too ideal although probably possible. I don't have the bandwidth to try and re-create it at the moment, but it could be done. You could also render a whole bunch of `Rectangles`, but I don't think the way the current axes work will bode well for having one value on multiple lines like the days of the week there

**Comment by vnugent** [comment]
I was thinking for the Y-axis, it could work by converting days of the week to 0 - 6 and custom render stacked bars with gaps.

**Reply by ckifer** [reply]
Feel free to take a stab at it and share if you get stuck (I don't have time atm but this would be a great example addition)

**Comment by vnugent** [comment]
After digging around (search keywords: filled squares) I found this issue https://github.com/recharts/recharts/issues/1068 which links to a scatter chart demo with square dots.  For whatever reason that demo is not shown anywhere on the website.

Basically, you'll need to use a scatter chart with data containing x, y, z (with z being the shade of the square), and use a custom renderer to paint all the squares.  (See issue #1068 for sample code).  The tricky part is calculating the chart aspect ratio (Width x Height) so that horizontal and vertical gaps are the same.

PS: I'd recommend using a date library such as https://date-fns.org/ to simplify date calculations.

Here's what I came up with so far:

<img width="1037" alt="Screenshot 2023-08-13 at 6 45 55 PM" src="https://github.com/recharts/recharts/assets/3805254/d500735b-445e-431c-b645-f4d37992f603">

**Reply by ckifer** [reply]
Sweet, we can add it to the storybook
